### PR TITLE
WIP: Convert firewalld to iptables

### PR DESCRIPTION
**Pull request by ddaskal**
I've changed all firewall implementations by firewalld to iptables.
I noticed NoVNC had a firewall rule allowing `udp` when it should be `tcp`, so fixed here.

### CODE REVIEW

File: roles/common/tasks/iptables.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  lineinfile:
+    dest: /etc/sysconfig/iptables
+    state: present
+    regexp: "^.*ACCEPT"
+    insertafter: "^.*ACCEPT"
+    line: '-A INPUT -m state --state NEW -m tcp -p tcp --dport 22 -j ACCEPT'
+    backup: yes
+  notify: restart iptables
+
+- name: Permit SSH For Sources
```
Comment (nirajdp76) [inline roles/common/tasks/iptables.yml]:
```
- name: Permit SSH
  lineinfile:
    dest: /etc/sysconfig/iptables
    state: present
    regexp: "^.*INPUT.*{{item.protocol}}.*{{ item.port }}.*ACCEPT"
    insertbefore: "^-A INPUT -j REJECT --reject-with icmp-host-prohibited$"
    line: '-A INPUT -m state --state NEW -m "{{ item.protocol }}" -p "{{ item.protocol }}" --dport "{{ item.port }}" -j ACCEPT'
    backup: yes
  with_items:
    - protocol: tcp
      port: 22
```

Comment (nirajdp76) [inline roles/common/tasks/iptables.yml]: Pretty much same comment for rest of the places where we writting the rules.

File: roles/common/tasks/iptables.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  lineinfile:
+    dest: /etc/sysconfig/iptables
+    state: present
+    regexp: "^.*ACCEPT"
```
Comment (nirajdp76) [inline roles/common/tasks/iptables.yml]: regexp: "^.*INPUT.*{{item.protocol}}.*-s {{ item.allow_form }}.*{{ item.port }}.*ACCEPT"

File: roles/common/tasks/iptables.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    line: '-A INPUT -m state --state NEW -m tcp -p tcp -s "{{ item }}" --dport 22 -j ACCEPT'
+    backup: yes
+  with_items: "{{ common.ssh.allow_from }}"
+  notify: restart iptables
+
+- name: Permit unrestricted access from remainder of cluster
+  lineinfile:
+    dest: /etc/sysconfig/iptables
+    state: present
+    regexp: "^.*ACCEPT"
```
Comment (nirajdp76) [inline roles/common/tasks/iptables.yml]: regexp: "^.INPUT.{{item.protocol}}.-s {{ item.allow_form }}.*ACCEPT"

File: roles/common/tasks/iptables.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+- name: Permit unrestricted access from remainder of cluster
+  lineinfile:
+    dest: /etc/sysconfig/iptables
+    state: present
+    regexp: "^.*INPUT.*{{ item.protocol }}.*-s {{ item.source }}.*ACCEPT"
+    insertbefore: "^-A INPUT -j REJECT --reject-with icmp-host-prohibited"
+    line: "-A INPUT -m state --state NEW -m {{ item.protocol }} -p {{ item.protocol }} -s {{ item.source.cidr }} -j ACCEPT"
+    backup: yes
+  with_items:
+    - source: "{{ undercloud_cidr }}"
```
Comment (nirajdp76) [inline roles/common/tasks/iptables.yml]: This is a list so ur logic in regex and line above will not work

File: roles/common/tasks/iptables.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+- name: Permit SSH For Sources
+  lineinfile:
+    dest: /etc/sysconfig/iptables
+    state: present
+    regexp: "^.*INPUT.*{{ item.protocol }}.*-s {{ item.source }}.*{{ item.port }}.*ACCEPT"
+    insertbefore: "^-A INPUT -j REJECT --reject-with icmp-host-prohibited"
+    line: "-A INPUT -m state --state NEW -m {{ item.protocol }} -p {{ item.protocol }} -s {{ item.source }} --dport {{ item.port }} -j ACCEPT"
+    backup: yes
+  with_items:
+    - source: "{{ common.ssh.allow_from }}"
```
Comment (nirajdp76) [inline roles/common/tasks/iptables.yml]: This is a list so ur logic in regex and line above will not work

Comment (nirajdp76) [inline roles/common/tasks/iptables.yml]: change it to
```
- name: Permit SSH For Sources
  lineinfile:
    dest: /etc/sysconfig/iptables
    state: present
    regexp: "^.*INPUT.*tcp.*-s {{ item }}.*22.*ACCEPT"
    insertbefore: "^-A INPUT -j REJECT --reject-with icmp-host-prohibited"
    line: "-A INPUT -m state --state NEW -m tcp -p tcp -s {{ item }} --dport {{ item }} -j ACCEPT"
    backup: yes
  with_items: "{{ common.ssh.allow_from }}"
  notify: restart iptables
```

File: roles/common/tasks/iptables.yml (hunk 4 above)
Comment (nirajdp76) [inline roles/common/tasks/iptables.yml]: Change it to
```
- name: Permit unrestricted access from remainder of cluster
  lineinfile:
    dest: /etc/sysconfig/iptables
    state: present
    regexp: "^.*INPUT.*all.*-s {{ item.cidr }}.*ACCEPT"
    insertbefore: "^-A INPUT -j REJECT --reject-with icmp-host-prohibited"
    line: "-A INPUT -m state --state NEW -m all -p all -s {{ item.cidr }} -j ACCEPT"
    backup: yes
  with_items: "{{ undercloud_cidr }}"
  notify: restart iptables
  when: undercloud_cidr is defined
```

File: roles/common/tasks/iptables.yml (hunk 5 above)
Comment (panxia6679) [inline roles/common/tasks/iptables.yml]: It failed when running on rhel vms, and here is my solution.

- name: Permit SSH For Sources
  lineinfile:
    dest: /etc/sysconfig/iptables
    state: present
    regexp: "^.*INPUT.*{{ item[1] }}.*-s {{ item[0]}}.*{{ item[2] }}.*ACCEPT"
    insertbefore: "^-A INPUT -j REJECT --reject-with icmp-host-prohibited"
    line: "-A INPUT -m state --state NEW -m {{ item[1] }} -p {{ item[1] }} -s {{ item[0] }} --dport {{ item[2] }} -j ACCEPT"
    backup: yes
  with_nested:
    - "{{ common.ssh.allow_from }}"
    - tcp
    - 22
  notify: restart iptables

File: roles/common/tasks/iptables.yml (hunk 4 above)
Comment (panxia6679) [inline roles/common/tasks/iptables.yml]: It failed as well. My solution:

~~~
- name: Permit unrestricted access from remainder of cluster
  lineinfile:
    dest: /etc/sysconfig/iptables
    state: present
    regexp: "^.*INPUT.*{{ item[1] }}.*-s {{ item[0].cidr }}.*ACCEPT"
    insertbefore: "^-A INPUT -j REJECT --reject-with icmp-host-prohibited"
    line: "-A INPUT -m state --state NEW -p {{ item[1] }} -s {{ item[0].cidr }} -j ACCEPT"
    backup: yes
  with_nested:
    - "{{ undercloud_cidr }}"
    - all
  notify: restart iptables
  when: undercloud_cidr is defined
~~~

**Comment by panxia6679** [comment]
After correcting these two places above, I can successfully deploy it on rhel VMs.

Here is content of iptables after deployment (after several failure).

<script src="https://gist.github.com/panxia6679/5c3d4e9a153a7375ee710d653ca05b49.js"></script>

**Comment by nirajdp76** [comment]
https://github.com/blueboxgroup/ursula/pull/2517

**Pull request closed without merging**